### ISSUE TITLE
Error while building with IP-Glasma

### ISSUE THREAD

**Issue by skhade01** [JETSCAPE/JETSCAPE#149]
Dear Experts,
 I am using JETSCAPE version 3.5.4. I have successfully built JETSCAPE with external packages like SMASH, MUSIC, etc. But I am facing the following error while making JETSCAPE with IPGlasma. 

  **CMake Error at /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find FFTW (missing: FFTW_INCLUDE_DIRS)**

I use JETSCAPE in a docker container. I installed FFTW in my system (Not in the docker container), but it didn't solve my issue.

Thank you.

**Comment by latessa** [comment]
Hi.  For your reference, FFTW has now been added to the jetscape/base:stable Docker image.

**Comment by LSchwiebert** [comment]
Can you please let us know if your issue has been resolved?

**Comment by jhputschke** [comment]
Close the issue.

**Comment by skhade01** [comment]
Hi Experts,
Apologies for the late reply. I was on leave and had a poor internet connection.
The error of **"CMake Error at /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
Could NOT find FFTW (missing: FFTW_INCLUDE_DIRS)"** is still there.

Just to avoid conflict with any previous versions, I reinstalled Docker and tried to build with a new jetscape/base:stable Docker image, but no luck.

I even tried to build Jetscape with IP-Glasma on a completely new machine yesterday, but facing the same issue.

Is it possible to discuss on this thread or should I initiate a new issue?

Thank you.

**Comment by jhputschke** [comment]
I reopened the issue so that we can follow up discussions in this thread ...

**Comment by latessa** [comment]
Replace the file JETSCAPE/external_packages/ipglasma/CMakeModules/FindFFTW.cmake with this alternative FindFFTW.cmake found here:

https://git.jinr.ru/nica/bmnroot/-/blob/9fb98e26eb3e27fe379d3a61bad5d1567665bd81/cmake/modules/FindFFTW.cmake

Then create a file called fftw3.h in the JETSCAPE/external_packages/ipglasma/src folder that includes the contents of the file linked here:

https://github.com/FFTW/fftw3/blob/master/api/fftw3.h

Then include the fftw library paths as part of your cmake command:

```
cmake .. -DCMAKE_CXX_STANDARD=14 -DUSE_ROOT=ON -DUSE_MUSIC=ON -DUSE_ISS=ON -DUSE_FREESTREAM=ON -DUSE_SMASH=ON -DUSE_IPGlasma=ON -DFFTW_INCLUDE_DIR=/usr/lib/x86_64-linux-gnu/libfftw3.so.3.5.8 -DFFTW_LIBRARY=/usr/lib/x86_64-linux-gnu/libfftw3.so.3.5.8
```
If you're not compiling all the external modules, you don't have to turn them on.

Doing this, I was able to compile JETSCAPE 3.5.4 with ipglasma using the stable Docker image.  Others who know more about ipglasma and fftw than I can comment about whether a better or more permanent solution can be implemented, but I'm offering this as a temporary solution in which I was able to successfully compile.
